src/anomalib/pre_processing/utils/transform.py

```diff
@@ -0,0 +1,200 @@
+"""Utility functions for transforms."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from collections.abc import Sequence
+
+from torch.utils.data import DataLoader
+from torchvision.transforms.v2 import CenterCrop, Compose, Resize, Transform
+
+from anomalib.data import AnomalibDataModule
+from anomalib.data.transforms import ExportableCenterCrop
+
+
+def get_stage_transform(stage: str, transforms: dict[str, Transform | None]) -> Transform | None:
+    """Get the transform for a specific stage.
+
+    Args:
+        stage: The stage to get the transform for (fit, validate, test, predict).
+        transforms: Dictionary mapping stage names to transforms.
+
+    Returns:
+        Transform for the specified stage, or None if not found.
+    """
+    stage_transforms_mapping = {
+        "fit": transforms.get("train"),
+        "validate": transforms.get("val"),
+        "test": transforms.get("test"),
+        "predict": transforms.get("test"),  # predict uses test transform
+    }
+    return stage_transforms_mapping.get(stage)
+
+
+def get_datamodule_transforms(datamodule: AnomalibDataModule) -> dict[str, Transform] | None:
+    """Get transforms from datamodule if available.
+
+    Args:
+        datamodule: The datamodule to get transforms from.
+
+    Returns:
+        Dictionary of transforms if found in datamodule, None otherwise.
+    """
+    if hasattr(datamodule, "train_transform") and hasattr(datamodule, "eval_transform"):
+        return {
+            "train": datamodule.train_transform,
+            "val": datamodule.eval_transform,
+            "test": datamodule.eval_transform,
+        }
+    return None
+
+
+def get_dataloaders_transforms(dataloaders: Sequence[DataLoader]) -> dict[str, Transform]:
+    """Get transforms from dataloaders.
+
+    Args:
+        dataloaders: The dataloaders to get transforms from.
+
+    Returns:
+        Dictionary mapping stages to their transforms.
+    """
+    transforms: dict[str, Transform] = {}
+    stage_lookup = {
+        "fit": "train",
+        "validate": "val",
+        "test": "test",
+        "predict": "test",
+    }
+
+    for dataloader in dataloaders:
+        if not hasattr(dataloader, "dataset") or not hasattr(dataloader.dataset, "transform"):
+            continue
+
+        for stage in stage_lookup:
+            if hasattr(dataloader, f"{stage}_dataloader"):
+                transforms[stage_lookup[stage]] = dataloader.dataset.transform
+
+    return transforms
+
+
+def set_datamodule_transforms(datamodule: AnomalibDataModule, transforms: dict[str, Transform | None]) -> None:
```

Comment:
I think a less similar naming between this one and `set_datamodule_transform` would improve readability. Maybe something like `set_stage_transform` or similar.

Reply:
Renamed in https://github.com/openvinotoolkit/anomalib/pull/2358/commits/6ea536966c0a077c79ab87c3699db802b2a02bf7